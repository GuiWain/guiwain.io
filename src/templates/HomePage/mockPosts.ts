import { Tags } from 'components/CardPost'

export const posts = [
  {
    title: 'Nullam in nisl eget tellus bibendum.',
    slug: '1',
    description:
      'Aliquam felis lorem, viverra ac est at, ornare ultrices felis. Donec cursus purus id tempus posuere. Cras nec felis metus.',
    date: 'Novembro 10, 2022',
    reading_time: 10,
    tags: ['dev'] as Tags[]
  },
  {
    title: 'Morbi consequat dolor condimen consectetur consectetur. ',
    slug: '2',
    description:
      'Aliquam felis lorem, viverra ac est at, ornare ultrices felis. Donec cursus purus id tempus posuere. Cras nec felis metus.',
    date: 'Novembro 10, 2022',
    reading_time: 7,
    tags: ['travel'] as Tags[]
  },
  {
    title:
      'Nullam in nisl eget tellus bibendum. Donec cursus purus id tempus posuere. ',
    slug: '3',
    description:
      'Aliquam felis lorem, viverra ac est at, ornare ultrices felis. Donec cursus purus id tempus posuere. Cras nec felis metus.',
    date: 'Novembro 10, 2022',
    reading_time: 4,
    tags: ['life'] as Tags[]
  }
]
