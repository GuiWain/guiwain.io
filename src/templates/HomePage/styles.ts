import styled, { css } from 'styled-components'

import UseCards from 'components/CardPost'

export const Wrapper = styled.div`
  width: 90%;
  margin: auto;
`

export const ListPosts = styled.section`
  margin: 3rem auto;

  display: flex;
  flex-direction: column;
  gap: 5rem;
`

export const CardPost = styled(UseCards)``

export const Description = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family02};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.regular};

    text-align: center;
    margin: 4rem auto;
    width: 70%;
  `}
`
