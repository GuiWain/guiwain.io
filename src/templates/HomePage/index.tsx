import * as S from './styles'

const HomePage = () => (
  <S.Wrapper>
    <S.Description>
      Quisque consectetur lacinia finibus. Pellentesque euismod risus
    </S.Description>
  </S.Wrapper>
)

export default HomePage
