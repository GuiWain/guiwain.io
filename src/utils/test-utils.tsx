import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'
import * as theme from 'styles/theme'

const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme.light}>{children}</ThemeProvider>)

export * from '@testing-library/react'
export { renderWithTheme as render }
