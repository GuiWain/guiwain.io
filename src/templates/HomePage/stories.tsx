import { ComponentMeta, ComponentStory } from '@storybook/react'
import HomePage from '.'

export default {
  title: 'Components/HomePage',
  component: HomePage
} as ComponentMeta<typeof HomePage>

export const Default: ComponentStory<typeof HomePage> = () => <HomePage />
