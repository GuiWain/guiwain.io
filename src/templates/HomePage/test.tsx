import { render } from 'utils/test-utils'

import HomePage from '.'

describe('<HomePage />', () => {
  it('should render the HomePage', () => {
    const { container } = render(<HomePage />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
