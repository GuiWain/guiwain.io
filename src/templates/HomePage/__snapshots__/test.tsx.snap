// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render the HomePage 1`] = `
.c0 {
  color: #2D2529;
  font-family: Merriweather,serif;
  font-size: 1.6rem;
  font-weight: 400;
  text-align: center;
  margin: 1rem auto 0;
  width: 80%;
}

<div
  class=""
>
  <h1
    class="c0"
  >
    Quisque consectetur lacinia finibus. Pellentesque euismod risus
  </h1>
</div>
`;
